src/legacy/ui/public/agg_types/metrics/top_hit.js

```diff
@@ -36,10 +36,9 @@ aggTypeFieldFilters.addFilter(
   (
     field,
     fieldParamType,
-    aggConfig,
-    vis
+    aggConfig
   ) => {
-    if (aggConfig.type.name !== 'top_hits' || vis.type.name === 'table' || vis.type.name === 'metric') {
+    if (aggConfig.type.name !== 'top_hits' || _.get(aggConfig.schema, 'aggSettings.top_hits.allowStrings', false)) {
```

Comment:
@timroes 
IMHO, it should be ` _.get(aggConfig.schema, 'aggSettings.top_hits.allowStrings', true)` in this condition, no?

Reply:
Sorry, forget my previous comment!
I thought that `false` was expected value, while it is the default value...

Reply:
By default, we don't want to allow strings, since we would assume that `top_hits` as a metric visualization should return numeric values like all other metric visualizations. So I would like everyone to by default need to enable this, if not this will be false, thus you won't jump into the `if` and return only numeric fields after the `if` clause.

Reply:
Absolutely, you're right!